docs/resources/server/recipes/builtin.md

```diff
@@ -0,0 +1,368 @@
+  }
+}
+```
+
+Let's digest this line for line:
+
+- `type`: This is the id of the shaped recipe serializer, `minecraft:crafting_shaped`.
+- `category`: This optional field defines the category in the crafting book.
+- `key` and `pattern`: Together, these define how the items must be put into the crafting grid.
+  - The pattern defines up to three lines of up to three-wide strings that define the shape. All lines must be the same length, i.e. the pattern must form a rectangular shape. Spaces can be used to denote slots that should stay empty.
+  - The key associates the characters used in the pattern with [ingredients][ingredient]. In the above example, all `X`s in the pattern must be iron ingots, and all `#`s must be sticks.
+- `result`: The result of the recipe. This is [an item stack's JSON representation][itemjson].
+- Not shown in the example is the `group` key. This optional string property creates a group in the recipe book. Recipes in the same group will be displayed as one in the recipe book.
+
+And then, let's have a look at how you'd generate this recipe:
+
+```java
+// We use a builder pattern, therefore no variable is created. Create a new builder by calling
+// ShapedRecipeBuilder#shaped with the recipe category (found in the RecipeCategory enum)
+// and a result item, a result item and count, or a result item stack.
+ShapedRecipeBuilder.shaped(RecipeCategory.EQUIPMENT, Items.IRON_PICKAXE)
+        // Create the lines of your pattern. Each call to #pattern adds a new line.
+        // Patterns will be validated, i.e. their shape will be checked.
+        .pattern("XXX")
+        .pattern(" # ")
+        .pattern(" # ")
+        // Create the keys for the pattern. All non-space characters used in the pattern must be defined.
+        // This can either accept Ingredients, TagKey<Item>s or ItemLikes, i.e. items or blocks.
+        .define('X', Items.IRON_INGOT)
+        .define('#', Items.STICK)
+        // Creates the recipe advancement. While not mandated by the consuming background systems,
+        // the recipe builder will crash if you omit this. The first parameter is the advancement name,
+        // and the second one is the condition. Normally, you want to use the has() shortcut for the condition.
+        // Multiple advancement requirements can be added by calling #unlockedBy multiple times.
+        .unlockedBy("has_iron_ingot", has(Items.IRON_INGOT))
+        // Stores the recipe in the passed RecipeOutput, to be written to disk.
+        // If you want to add conditions to the recipe, those can be set on the output.
+        .save(output);
+```
+
+Additionally, you can call `#group` to set the recipe book group.
+
+### Shapeless Crafting
+
+Unlike shaped crafting recipes, shapeless crafting recipes do not care about the order the ingredients are passed in. As such, there is no pattern and key, instead there is just a list of ingredients:
+
+```json5
+{
+  "type": "minecraft:crafting_shapeless",
+  "category": "misc",
```

Comment:
```suggestion
  "category": "equipment",
```
See the comment above, equipment would be the correct `CraftingBookCategory` (which is what ends up in the JSON) for `RecipeCategory.COMBAT`